jax/numpy/lax_numpy.py

```diff
@@ -597,19 +597,19 @@ def angle(x):
 
 @_wraps(onp.reshape)
 def reshape(a, newshape, order="C"):  # pylint: disable=missing-docstring
-  if order == "C" or order is None:
-    dims = None
+  dummy_val = onp.broadcast_to(0, shape(a))  # zero strides
+  computed_newshape = onp.reshape(dummy_val, newshape).shape
+
+  if order == "C":
+    return lax.reshape(a, computed_newshape, None)
   elif order == "F":
     dims = onp.arange(ndim(a))[::-1]
+    return lax.reshape(a, computed_newshape[::-1], dims).T
```

Comment:
Personally, I'd forget the third argument to XLA's reshape operator even exists (and we should remove it from LAX). It expands into a reshape and a transpose anyway.

Just write this as:
`lax.reshape(a.T, computed_newshape[::-1]).T`
?

Reply:
Hm we could do that, but actually as written it's slightly better for op-by-op dispatch, because even if something expands inside HLO to a composition of two HLO, that's better than incurring two dispatches in the op-by-op setting.

Do you buy that?

Reply:
I think that's mostly microoptimizing and if you think it's important you should stick a `jit` decorator on the function :-)

Reply:
I hear you, but since that change seems like a low priority, I'm going to merge this so as to reduce the number of balls in the air right now.